tests/storage/restricted_namespace_cloning/utils.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+import pytest
+from kubernetes.client.rest import ApiException
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.datavolume import DataVolume
+
+from tests.storage.restricted_namespace_cloning.constants import TARGET_DV
+from tests.storage.utils import assert_pvc_snapshot_clone_annotation
+from utilities.constants import PVC
+from utilities.storage import (
+    ErrorMsg,
+    create_dv,
+    is_snapshot_supported_by_sc,
+    sc_volume_binding_mode_is_wffc,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def verify_snapshot_used_namespace_transfer(cdv: DataVolume, unprivileged_client: DynamicClient) -> None:
+    storage_class = cdv.storage_class
+    # Namespace transfer is not possible with WFFC
+    if is_snapshot_supported_by_sc(
+        sc_name=storage_class, client=unprivileged_client
+    ) and not sc_volume_binding_mode_is_wffc(sc=storage_class):
```

Comment:
What would be the verification if the condition is not met? e.g. test_create_vm_with_cloned_data_volume_positive(), this is the only verification. Should that test run if storage class does not support snapshot and volume binding is wffc?

Reply:
Ability to create a vm should be a separate test, no?

Reply:
The tests should be separate. Two verifications should not be combined.

Reply:
This has always been the process in CNV. We review PRs as a whole, instead of considering just newly introduced code. While it might be a bit frustrating for the person enhancing/reusing the code, it also gives us opportunity to improve things and bring different perspectives into our consideration. The work you did with run time reduction is significant, I can only hope that you would take the review comments as an opportunity to go further and improve the test and utilities more.

Reply:
Please handle it at some point. Specially when you work on removing the programmatic skips from storage lane.

Reply:
Sure, it will be good to handle this together with removing the programmatic skips